app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/supportingevidence/SupportingEvidenceController.scala

```diff
@@ -141,47 +131,40 @@ class SupportingEvidenceController @Inject() (
       Ok(scanFailedPage())
     }
 
-  def scanProgress(
-    uploadReference: UploadReference
-  ): Action[AnyContent] =
+  def scanProgress(uploadReference: UploadReference): Action[AnyContent] =
     authenticatedActionWithSessionData.async { implicit request =>
       withUploadSupportingEvidenceAnswers { (_, fillingOutReturn, answers) =>
-        answers match {
-          case _: CompleteSupportingEvidenceAnswer =>
-            Redirect(routes.SupportingEvidenceController.checkYourAnswers())
-
-          case incompleteAnswers: IncompleteSupportingEvidenceAnswer =>
-            val result = for {
-              upscanUpload <- upscanService.getUpscanUpload(uploadReference)
-              _            <- upscanUpload.upscanCallBack match {
-                                case Some(upscanSuccess: UpscanSuccess) =>
-                                  storeUpscanSuccess(
-                                    upscanUpload,
-                                    upscanSuccess,
-                                    incompleteAnswers,
-                                    fillingOutReturn
-                                  )
-                                case _                                  =>
-                                  EitherT.pure[Future, Error](())
-                              }
-            } yield upscanUpload
-
-            result.fold(
-              e => {
-                logger.warn(s"could not update the status of upscan upload to uploaded : $e")
-                errorHandler.errorResult()
-              },
-              upscanUpload =>
-                upscanUpload.upscanCallBack match {
-                  case Some(_: UpscanSuccess) =>
-                    Redirect(routes.SupportingEvidenceController.chooseSupportingEvidenceDocumentType(uploadReference))
-                  case Some(_: UpscanFailure) =>
-                    Redirect(routes.SupportingEvidenceController.handleUpscanCallBackFailures())
-                  case None                   =>
-                    Ok(scanProgressPage(upscanUpload))
-                }
-            )
-        }
+        val result = for {
+          upscanUpload <- upscanService.getUpscanUpload(uploadReference)
+          _            <- upscanUpload.upscanCallBack match {
+                            case Some(upscanSuccess: UpscanSuccess)
+                                if answers.forall(_.forall(_.uploadReference =!= uploadReference)) =>
```

Comment:
Avoiding adding duplicate by checking whether document already exists - in case if user hits page multiple times. On the next stage - whether document was added or not, will ask to amend existing document type.